presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileWriter.java

```diff
@@ -196,24 +207,24 @@ private static RecordWriter createRecordWriter(Path target, Configuration conf)
 
     private static List<ObjectInspector> getJavaObjectInspectors(List<StorageType> types)
     {
-        return types.stream().map(OrcFileWriter::getJavaObjectInspector).collect(toList());
+        return types.stream().map(StorageType::getHiveTypeName).map(TypeInfoUtils::getTypeInfoFromTypeString).map(OrcFileWriter::getJavaObjectInspector).collect(toList());
     }
 
-    private static ObjectInspector getJavaObjectInspector(StorageType type)
+    private static ObjectInspector getJavaObjectInspector(TypeInfo typeInfo)
     {
-        switch (type) {
-            case BOOLEAN:
-                return javaBooleanObjectInspector;
-            case LONG:
-                return javaLongObjectInspector;
-            case DOUBLE:
-                return javaDoubleObjectInspector;
-            case STRING:
-                return javaStringObjectInspector;
-            case BYTES:
-                return javaByteArrayObjectInspector;
-        }
-        throw new PrestoException(INTERNAL_ERROR, "Unhandled storage type: " + type);
+        Category category = typeInfo.getCategory();
+        if (category == PRIMITIVE) {
+            return getPrimitiveJavaObjectInspector(getPrimitiveTypeInfo(typeInfo.getTypeName()));
+        }
+        if (category == LIST) {
+            ListTypeInfo listTypeInfo = (ListTypeInfo) typeInfo;
+            return getStandardListObjectInspector(getJavaObjectInspector(listTypeInfo.getListElementTypeInfo()));
+        }
+        if (category == MAP) {
+            MapTypeInfo mapTypeInfo = (MapTypeInfo) typeInfo;
+            return getStandardMapObjectInspector(getJavaObjectInspector(mapTypeInfo.getMapKeyTypeInfo()), getJavaObjectInspector(mapTypeInfo.getMapValueTypeInfo()));
+        }
+        throw new PrestoException(INTERNAL_ERROR, "Unhandled storage type: " + typeInfo);
```

Comment:
Log the category instead of the `typeInfo` since that is what we are handling above